app/models/helpers/piece_helper.rb

```diff
@@ -9,9 +9,9 @@ def is_obstructed_horizontally?(row_destination, col_destination)
     end_column = col_destination + (incrementing ? -1 : 1)
 
     if incrementing
-      self.game.pieces.active.where(row: self.row, column: (start_column..end_column)).exists?
+      self.game.pieces.where(row: self.row, column: (start_column..end_column)).exists?

```

Comment:
So, here you will want to switch this back to using the active scope.
`self.game.pieces.active.where(row: self.row, column: (start_column..end_column)).exists?`
There are a couple of other places in piece_helper that you'll need to make the switch as well. 
Other than that the code looks good to go I think.